content/blog/2016/2016-04-07-pipeline-for-runs-on-hardware.adoc

```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: Automating test runs on hardware with Pipeline-as-Code. Part 1
+tags:
+- jenkins
+- jenkins2
+- pipeline
+- embedded
+author: oleg_nenashev
+---
+
+In addition to Jenkins development, during last 8 years I've been involved into continuous integration for hardware and embedded projects.
+At JUC2015/London 
+link:https://www.cloudbees.com/jenkins/juc-2015/presentations/JUC-2015-Europe-Jenkins-Based-CI-for-Nenashev.pdf[I have conducted a talk] about common automation challenges in the area.
+
+In this blog post I would like to concentrate on link:https://jenkins.io/doc/pipeline/[Pipeline] (fka Workflow), which is a new automation-as-code ecosystem for Jenkins.
+It will be proposed by default in the incoming link:/2.0/[Jenkins 2.0 release].
+
+First time I've tried Pipeline two and half years ago, and unfortunately it didn't work for my cases at all. 
+I was very disappointed and tried it only one year later, At this point the plugin became much more stable and useful. 
+It also attracted much more contributors, and it started evolving rapidly and finally became a huge ecosystem powered by Jenkins.
```

Comment:
> I was very disappointed and tried it only one year later, At this point the plugin became much more stable and useful. It also attracted much more contributors, and it started evolving rapidly and finally became a huge ecosystem powered by Jenkins.

I think I understand what you're trying to get across here, but I think the sentence is a little disjointed. Also "a huge ecosystem powered by Jenkins" doesn't quite make sense to me

How about:

> I was very disappointed but tried it again a year later. This time, the plugin had become much more stable and useful. It had also attracted more contributors and started evolving more rapidly with the development of plugins extending the Pipeline ecosystem.